internal/users/manager_test.go

```diff
@@ -173,25 +190,35 @@ func TestUpdateUser(t *testing.T) {
 			Name: "gpasswdfail",
 			GID:  nil,
 		}},
+		"nameless-group": {{
+			Name: "",
+			GID:  ptrValue(11111),
+		}},
 		"no-groups": {},
 	}
 
 	tests := map[string]struct {
+		userCase   string
 		groupsCase string
 
 		dbFile          string
 		localGroupsFile string
 
-		wantErr error
+		wantErr  error
+		noOutput bool
 	}{
 		"Successfully update user":                       {groupsCase: "cloud-group"},
 		"Successfully update user updating local groups": {groupsCase: "mixed-groups-cloud-first", localGroupsFile: "users_in_groups.group"},
 
-		"Error if no groups were provided":          {groupsCase: "no-groups", wantErr: shouldError{}},
-		"Error if only local group was provided":    {groupsCase: "local-group", wantErr: shouldError{}},
-		"Error if local group is the default group": {groupsCase: "mixed-groups-local-first", wantErr: shouldError{}},
+		"Error if user has no username":             {userCase: "nameless", wantErr: shouldError{}, noOutput: true},
+		"Error if group has no name":                {groupsCase: "nameless-group", wantErr: shouldError{}, noOutput: true},
+		"Error if no groups were provided":          {groupsCase: "no-groups", wantErr: shouldError{}, noOutput: true},
+		"Error if only local group was provided":    {groupsCase: "local-group", wantErr: shouldError{}, noOutput: true},
+		"Error if local group is the default group": {groupsCase: "mixed-groups-local-first", wantErr: shouldError{}, noOutput: true},
 
-		"Remove user from db if can't update local groups": {groupsCase: "mixed-groups-gpasswd-fail", localGroupsFile: "gpasswdfail_in_deleted_group.group", wantErr: shouldError{}},
+		"Remove user from db if can't update local groups":                 {groupsCase: "mixed-groups-gpasswd-fail", localGroupsFile: "gpasswdfail_in_deleted_group.group", wantErr: shouldError{}},
+		"Remove user from db with many users if can't update local groups": {dbFile: "multiple_users_and_groups", groupsCase: "mixed-groups-gpasswd-fail", localGroupsFile: "gpasswdfail_in_deleted_group.group", wantErr: shouldError{}},
+		"Remove existing user from db if can't update local groups":        {userCase: "user2", dbFile: "multiple_users_and_groups", groupsCase: "mixed-groups-gpasswd-fail", localGroupsFile: "gpasswdfail_in_deleted_group.group", wantErr: shouldError{}},
```

Comment:
Ditto, I would be a little bit more verbose too, like "keeps other users" or so, to really describe the effect on only failing on one user.